nntrainer/dataset/databuffer_func.cpp

```diff
@@ -86,14 +86,15 @@ int DataBufferFromCallback::init() {
 }
 
 int DataBufferFromCallback::setGeneratorFunc(DatasetDataUsageType type,
-                                             datagen_cb func) {
+                                             datagen_cb func, void *user_data) {
 
   int status = ML_ERROR_NONE;
   switch (type) {
   case DatasetDataUsageType::DATA_TRAIN:
     if (!func)
       return ML_ERROR_INVALID_PARAMETER;
     callback_train = func;
```

Comment:
I am assuming the dependency on the type of dataset this is (train/valid/test) will be removed in future PRs.

Reply:
Correct 👍